extensions/opencv/CaptureRTSPFrame.cpp

```diff
@@ -0,0 +1,170 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CaptureRTSPFrame.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+static core::Property rtspUsername;
+static core::Property rtspPassword;
+static core::Property rtspHostname;
+static core::Property rtspURI;
+static core::Property captureFrameRate;
+static core::Property imageEncoding;
+
+core::Property CaptureRTSPFrame::RTSPUsername(  // NOLINT
+    "RTSP Username",
+    "The username for connecting to the RTSP stream", "");
+core::Property CaptureRTSPFrame::RTSPPassword(  // NOLINT
+    "RTSP Password",
+    "Password used to connect to the RTSP stream", "");
+core::Property CaptureRTSPFrame::RTSPHostname(  // NOLINT
+    "RTSP Hostname",
+    "Hostname of the RTSP stream we are trying to connect to", "");
+core::Property CaptureRTSPFrame::RTSPURI(  // NOLINT
+    "RTSP URI",
+    "URI that should be appended to the RTSP stream hostname", "");
+core::Property CaptureRTSPFrame::RTSPPort(  // NOLINT
+    "RTSP Port",
+    "Port that should be connected to to receive RTSP Frames",
+    "");
+core::Property CaptureRTSPFrame::ImageEncoding( // NOLINT
+    "Image Encoding",
+    "The encoding that should be applied the the frame images captured from the RTSP stream",
+    ".jpg"
+    );
+
+core::Relationship CaptureRTSPFrame::Success(  // NOLINT
+    "success",
+    "Successful capture of RTSP frame");
+core::Relationship CaptureRTSPFrame::Failure(  // NOLINT
+    "failure",
+    "Failures to capture RTSP frame");
+
+void CaptureRTSPFrame::initialize() {
+  std::set<core::Property> properties;
+  properties.insert(RTSPUsername);
+  properties.insert(RTSPPassword);
+  properties.insert(RTSPHostname);
+  properties.insert(RTSPPort);
+  properties.insert(RTSPURI);
+  properties.insert(ImageEncoding);
+  setSupportedProperties(std::move(properties));
+
+  std::set<core::Relationship> relationships;
+  relationships.insert(Success);
+  relationships.insert(Failure);
+  setSupportedRelationships(std::move(relationships));
+}
+
+void CaptureRTSPFrame::onSchedule(core::ProcessContext *context, core::ProcessSessionFactory *sessionFactory) {
+
+  std::string value;
+
+  if (context->getProperty(RTSPUsername.getName(), value)) {
+    rtsp_username_ = value;
+  }
+  if (context->getProperty(RTSPPassword.getName(), value)) {
+    rtsp_password_ = value;
+  }
+  if (context->getProperty(RTSPHostname.getName(), value)) {
+    rtsp_host_ = value;
+  }
+  if (context->getProperty(RTSPPort.getName(), value)) {
```

Comment:
we now support getting the port directly as an integer if you use withDefaultValue<uint16_t>(port)